src/libs/Trie/TrieNode.js

```diff
@@ -0,0 +1,10 @@
+
+class TrieNode {
+    constructor() {
+        this.children = {};
+        this.isLeaf = false;
```

Comment:
I don't think we need (or want) to store this data separately because it's essentially duplication of data. Instead, we should create a new member function:

```js
isLeaf() {
    return _.isEmpty(this.children);
}
```

Reply:
This class properity `isLeaf`makes a conflict because maybe an end node in the trie has children.

for example in this trie:

![Screenshot from 2022-10-11 15-21-17](https://user-images.githubusercontent.com/108357004/195120769-3254b938-eeff-498b-b17c-7fd3de305813.png)

`CAT`and `CATTLE` both are words in the trie. Although the `T` node below the `A` node is an end node it has a child as well.

Reply:
Oh nice you're totally right!

Reply:
Done, `isEndOfWord` is the optimal name.